tiledb/api/c_api_test_support/testsupport_capi_array.h

```diff
@@ -0,0 +1,201 @@
+/**
+ * @file tiledb/api/c_api_test_support/testsupport_capi_array.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines test support for the array section of the C API.
+ */
+
+#ifndef TILEDB_TESTSUPPORT_CAPI_ARRAY_H
+#define TILEDB_TESTSUPPORT_CAPI_ARRAY_H
+
+#include "testsupport_capi_context.h"
+#include "tiledb/api/c_api/array/array_api_external.h"
+#include "tiledb/api/c_api/array/array_api_internal.h"
+#include "tiledb/api/c_api/array_schema/array_schema_api_internal.h"
+#include "tiledb/api/c_api/attribute/attribute_api_internal.h"
+#include "tiledb/api/c_api/dimension/dimension_api_internal.h"
+#include "tiledb/api/c_api/domain/domain_api_internal.h"
+
+namespace tiledb::api::test_support {
+
+class ordinary_array_exception : public StatusException {
+ public:
+  explicit ordinary_array_exception(const std::string& message = "")
+      : StatusException("error creating test array; ", message) {
+  }
+};
+
+/**
+ * Base class for an ordinary array.
+ */
+class ordinary_array_without_schema {
+ private:
+  const char* array_uri = "unit_capi_array";
+
+ public:
+  ordinary_context context{};
+  tiledb_array_handle_t* array{nullptr};
```

Comment:
There's no advantage to using smart pointers here; it would simply be excess overhead. The point is to call `...free()` in the destructor. This function does that just fine.